js/mail.js

```diff
@@ -22,6 +22,52 @@ var Mail = {
 			$('#searchresults').hide();
 		}
 	},
+	BackGround: {
+		checkForNotifications: function() {
+			_.each(Mail.State.accounts, function (a) {
+				Mail.Communication.get(OC.generateUrl('apps/mail/accounts/{accountId}/folders', {accountId: a.accountId}), {
+					success: function (jsondata) {
+						var localAccount = Mail.State.folderView.collection.get(jsondata.id);
+						var folders = localAccount.get('folders');
+						_.each(jsondata.folders, function(f) {
+							var localFolder = folders.get(f.id);
+							if (f.uidvalidity !== localFolder.get('uidvalidity') ||
+								f.uidnext !== localFolder.get('uidnext')) {
+
+								// send notification
+								Notification.requestPermission(function() {
+									new Notification(
+										"New Mail arrived!",
```

Comment:
Can we here directly display sender and subject of the email? First line sender, second line subject, just like the message list.

We need to keep the notification short and sweet. »New mail« is evident from the notification itself. ;)

If there are multiple new emails, we could show it a bit different like:
```
5 new emails
from Thomas Müller, Christoph Wurst, …
```